src/main/scala/org/phenoscape/kb/Similarity.scala

```diff
@@ -233,6 +237,53 @@ object Similarity {
       } yield JaccardScore(Set(left, right), intersectionCount.toDouble / unionCount.toDouble)).toSeq
     }
 
+  // Output a boolean matrix as CSV
+  def matrixRendererFromMapOfMaps(mapOfMaps: Map[Any, Map[Any, Boolean]]) = {
+
+    val sortedKeys = mapOfMaps.keys.toList.map(_.toString).sorted
+    var resultString = "headers, " + sortedKeys.mkString(", ") //print column headers
+
+    for(x <- sortedKeys) yield {
+      resultString = resultString.concat("\n" + x)
+      for(y <- sortedKeys) yield mapOfMaps.get(x).get(y) match {
+        case true  => resultString = resultString.concat(", " + "1")
+        case false => resultString = resultString.concat(", " + "0")
```

Comment:
I wouldn't stick the commas in here. You can add them when you convert your list of values into a string with `.mkString()`.